planning/autoware_obstacle_cruise_planner/src/pid_based_planner/pid_based_planner.cpp

```diff
@@ -200,9 +200,17 @@ std::optional<PIDBasedPlanner::CruiseObstacleInfo> PIDBasedPlanner::calcObstacle
       calcDistanceToCollisionPoint(planner_data, obstacle.collision_points.front().point) +
       (obstacle.velocity - planner_data.ego_vel) * time_to_evaluate_rss_;
 
-    // calculate distance between ego and obstacle based on RSS
-    const double target_dist_to_obstacle = calcRSSDistance(
-      planner_data.ego_vel, obstacle.velocity, longitudinal_info_.safe_distance_margin);
+    const double target_dist_to_obstacle = [&]() -> double {
+      if (longitudinal_info_.safe_distance_method == "time_based") {
+        // calculate distance between ego and obstacle based on time-based rule
+        return calcTimeBasedSafeDistance(
+          planner_data.ego_vel, longitudinal_info_.safe_distance_time_margin);
+      } else {
+        // calculate distance between ego and obstacle based on RSS
+        return calcRSSDistance(
+          planner_data.ego_vel, obstacle.velocity, longitudinal_info_.safe_distance_margin);
+      }
+    }();
```

Comment:
Can you create a member function of calcSafeDistance (you can change the name), and make calcTimeBasedSafeDistance and calcRSSDistance functions private so that other people won't use calcTimeBasedSafeDistance and calcRSSDistance directly by mistake?